internal/cluster/client_message_test.go

```diff
@@ -0,0 +1,152 @@
+package cluster
+
+import (
+	"bytes"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+)
+
+func TestClientMessageReader(t *testing.T) {
+	testCases := []struct {
+		f    func(t *testing.T)
+		name string
+	}{
+		{name: "readSingleFrame", f: readSingleFrame},
+		{name: "readMultiFrameMessage", f: readMultiFrameMessage},
+		{name: "readFramesInMultipleCallsToRead", f: readFramesInMultipleCallsToRead},
+		{name: "readWhenTheFrameLengthAndFlagsNotReceivedAtFirst", f: readWhenTheFrameLengthAndFlagsNotReceivedAtFirst},
+		{name: "readFramesInMultipleCallsToReadFromWhenLastPieceIsSmall", f: readFramesInMultipleCallsToReadFromWhenLastPieceIsSmall},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			tc.f(t)
+		})
+	}
+}
+
+func readSingleFrame(t *testing.T) {
+	frame := createFrameWithRandomBytes(t, 42)
+	message := proto.NewClientMessage(frame)
+	buffer := writeToBuffer(t, message)
+	reader := newClientMessageReader()
+	reader.Append(buffer.Bytes())
+	require.NotNil(t, reader.Read())
+	iterator := proto.NewForwardFrameIterator(reader.clientMessage.Frames)
+	require.True(t, iterator.HasNext())
+	frameRead := iterator.Next()
+	require.Equal(t, frame.Content, frameRead.Content)
+	require.False(t, iterator.HasNext())
+}
+
+func readMultiFrameMessage(t *testing.T) {
+	frame1 := createFrameWithRandomBytes(t, 10)
+	frame2 := createFrameWithRandomBytes(t, 20)
+	frame3 := createFrameWithRandomBytes(t, 30)
+	message := proto.NewClientMessageForEncode()
+	message.AddFrame(frame1)
+	message.AddFrame(frame2)
+	message.AddFrame(frame3)
+	buffer := writeToBuffer(t, message)
+	reader := newClientMessageReader()
+	reader.Append((*buffer).Bytes())
+	require.NotNil(t, reader.Read())
+	iterator := reader.clientMessage.FrameIterator()
+	require.True(t, iterator.HasNext())
+	frameRead := iterator.Next()
+	require.Equal(t, frame1.Content, frameRead.Content)
+	require.True(t, iterator.HasNext())
+	frameRead = iterator.Next()
+	require.Equal(t, frame2.Content, frameRead.Content)
+	require.True(t, iterator.HasNext())
+	frameRead = iterator.Next()
+	require.Equal(t, frame3.Content, frameRead.Content)
+	require.False(t, iterator.HasNext())
+}
+
+func readFramesInMultipleCallsToRead(t *testing.T) {
+	frame := createFrameWithRandomBytes(t, 1000)
+	message := proto.NewClientMessage(frame)
```

Comment:
```suggestion
	message := proto.NewClientMessageForEncode()
	message.AddFrame(frame)
```
As I explained above, It would be more meaningful to be comply with Java Client tests.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-go-client/pull/863/commits/73051a7840e4f303b95e8fd0abc090403aff37b6.